resources/js/features/game-list/components/WantToPlayGamesDataTable/DataTableSearchInput/DataTableSearchInput.tsx

```diff
@@ -0,0 +1,79 @@
+import type { Table } from '@tanstack/react-table';
+import { useEffect, useState } from 'react';
+import { useDebounce } from 'react-use';
+
+import { BaseInput } from '@/common/components/+vendor/BaseInput';
+import { cn } from '@/utils/cn';
+
+import { useSearchInputHotkey } from './useSearchInputHotkey';
+
+interface DataTableSearchInputProps<TData> {
+  table: Table<TData>;
+}
+
+export function DataTableSearchInput<TData>({ table }: DataTableSearchInputProps<TData>) {
+  const initialValue = (table.getColumn('title')?.getFilterValue() as string) ?? '';
+
+  const [rawInputValue, setRawInputValue] = useState(initialValue);
+
+  const { hotkeyInputRef } = useSearchInputHotkey({ key: '/' });
+
+  /**
+   * Listen for changes with column filter state and stay in sync. Otherwise,
+   * when the user presses the "Reset" button to reset all filters, our search
+   * value will remain. It needs to be reset too.
+   */
+  useEffect(() => {
+    const filterValue = (table.getColumn('title')?.getFilterValue() as string) ?? '';
+    setRawInputValue(filterValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- this is a valid dependency array
+  }, [table.getState().columnFilters]);
+
+  /**
+   * Wait until the user is done typing before we fetch from the back-end.
+   */
+  useDebounce(
+    () => {
+      if (rawInputValue.length >= 3 || rawInputValue.length === 0) {
+        table.getColumn('title')?.setFilterValue(rawInputValue);
+      }
```

Comment:
this behavior is a little weird. I typed in a three character string and got zero matches, then started backspacing to get some close matches, and the list didn't update until the search string was gone. then I started typing in the shorter matches without getting any matches, and at the third character, the list evaporated again.

I know it's not performant to do filters on small substrings - especially as you're typing, but the same could be said at 4 or five characters. If I'm typing in "dragon", it should either start filtering immediately at "D", or wait until I stop pushing keys for a second or so.

Reply:
Agreed, that can feel a bit strange.

I've tried to tweak this implementation in latest to (hopefully) improve the UX. Now, searches work regardless of input length. However, the shorter the input length, the longer the debounce.

```ts
useDebounce(
  () => {
    table.getColumn('title')?.setFilterValue(rawInputValue);
  },
  getDebounceDuration(rawInputValue),
  [rawInputValue],
);

function getDebounceDuration(input: string): number {
  // Don't debounce at all when the input is cleared.
  if (input.length === 0) {
    return 0;
  }

  // Use a longer debounce for shorter inputs.
  if (input.length < 3) {
    return 500;
  }

  // Use a shorter debounce for longer inputs.
  return 200;
}
```

Reply:
I'd probably go 1000 and 500. When typing in "dragon", i saw filters occurring for "dra", "drag", "drago", and "dragon", and I consider myself an above-average typist.

Reply:
Bumped to 1000 and 500 in https://github.com/RetroAchievements/RAWeb/pull/2706/commits/363ff75d6913a037182fd3c5f432b0e8c504ef01.